pkg/storage/boltdb/boltdb.go

```diff
@@ -0,0 +1,265 @@
+package boltdb
+
+import (
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"go.etcd.io/bbolt"
+	"zotregistry.io/zot/errors"
+	zlog "zotregistry.io/zot/pkg/log"
+	"zotregistry.io/zot/pkg/storage/cachedb"
+	"zotregistry.io/zot/pkg/storage/constants"
+)
+
+type Driver struct {
+	rootDir     string
+	db          *bbolt.DB
+	log         zlog.Logger
+	useRelPaths bool // weather or not to use relative paths, should be true for filesystem and false for s3
+}
+
+type DriverParameters struct {
+	RootDir, Name string
+	UseRelPaths   bool
+}
+
+const (
+	defaultRootDir     = "/tmp/zot/boltDB"
```

Comment:
I don't really like this path. Can we have one relative to the root dir? We had one in the old implementation.

Reply:
Removed the default values as they aren't really ever used. From the controller func CreateCacheDatabaseDriver, if an incomplete config is used, an errors.ErrBadConfig with nil driver is returned. If no configuration is provided, it will default to the storage rootDir.